### ISSUE TITLE
query request: all records made by X

### ISSUE THREAD

**Issue by lbestock** [arch-kiosk/arch-kiosk-office#3009]
We discussed the other day that often one remembers who did something, not what unit it was in or such. And today in a real world situation I wished for there to be a query that returned me all records associated with (or at least made by) a particular excavator. So I submit a request!

**Comment by urapadmin** [comment]
"all records" is a bit vague. What would you want the query to return in the list? The Identifier of the record and date? Perhaps the unit or site id, too?

**Comment by lbestock** [comment]
Identifier of the record, site and unit, date sounds good. In fact I would expect the results to look precisely like what #2648 gets, just it’s a person not a timespan in the query.

**Comment by lbestock** [comment]
If this were rolled out today in Abydos I would use it today....

**Comment by lbestock** [comment]
Oh cool, it just now occurs to me that this is in fact two separate queries. Because with legacy data, there will be a difference between the person who made the original paper record and the logged in creator of the digital record. And in fact I want queries that address BOTH of those - they answer different questions for me.

**Comment by urapadmin** [comment]
we don't have a person who originally made the record. At least not as a general thing per record. We have those only in a few records. That is really a design feature we missed out on in the very beginning. On the other hand, it is only needed for legacy records and I still feel we are misusing Kiosk here for something it was indeed not designed to do.

**Comment by lbestock** [comment]
We have it for both locus and narrative, well and also for OP, which is helpful. 

But I take your point about legacy data. Interestingly the task I want to accomplish right now would benefit from the person who entered it query that would be the more usually useful for non-legacy data. I want to check over a span of time, not day by day, on what students have been doing. Per student. Director's view for a month, not a day.

**Comment by urapadmin** [comment]
That is not possible. At least not with precision. We do not have a logging database (thank god) and so we always only have the last user changing something. If another user has changed the same thing in between that information would be lost. I think that is not super often the case but one needs to keep that in mind.

I think we should indeed separate these concerns. I can make you a query with a time span and a user and it shows you what identifier has been touched by that user (not what exactly the user did). That query would work with the modified_by field that all our records have.

And I could make you a separate, ANC specific query that addresses the manually assigned user-ids for unit, locus, cm. Again the result would be identifiers mostly (see #2648)

**Comment by urapadmin** [comment]
by the way if you have a specific ANC question that you only ask once, I can always query the database manually

**Comment by lbestock** [comment]
I will want to ask it again weekly more or less. 

The ANC query is for each of the three students in NY, to see what they have touched since we first started. (Which, since I think it's super rare for someone to mess with someone else's record, I fail to remember that we don't have a logging database.)

**Comment by urapadmin** [comment]
### Kiosk 1.6.24
- Normally a new query does not need a new Kiosk version. But this one does because I had to solve a bunch of open Q&V issues to make this possible: #3139 , #3140 , #2304 
- So this here is also the test for those tickets. They are otherwise untestable for common mortals
- This  still has limitations, the biggest being that you have to fill all fields with something reasonable. If you leave one blank or type in nonsense you'll get nasty server responses. Addressed by #3141

**Comment by luizaogs** [comment]
So this new query in 1.6.24 is for the actual excavator, not the student inputting things, yes? Do we need a new ticket for the latter? (I haven't checked if this is working yet, but if I close it.)

**Comment by luizaogs** [comment]
Um I may be using this incorrectly but I am trying to test this and I put in the following dates for Emily Russo, expecting at least Op 172 to come up since it was opened on 21.XII.2010. But I get nothing. 

<img width="427" alt="Image" src="https://github.com/user-attachments/assets/f8956127-3cb1-48e5-bd4a-8fa752a0d39f" />

<img width="1454" alt="Image" src="https://github.com/user-attachments/assets/a5222b87-0486-4530-bd55-a04694fa8192" />

**Comment by urapadmin** [comment]
yes, that is not what is is doing. This is working like the director's view. It is to see who worked on what recently in ANC.  It is addressing the time when the technical record was last modified and the user logged in to the recording app who modified it. 

That is different from when the archaeological record was created and who created it (the manually stated excavator). Such a query will follow.

**Comment by urapadmin** [comment]
perhaps different wording in the query description? There is a keyboard induced typo in the description, anyhow. I worry about you.

**Comment by luizaogs** [comment]
Yeah I missed that typo. Maybe you should find a new tester :)

**Comment by luizaogs** [comment]
But then maybe don't call it "list archaeological entities DUG by a certain team member"? That's what tripped me up.

**Comment by luizaogs** [comment]
Ok, so yes this is working: I chose May and random dates and get a million records. But I would definitely change the query name and description.

**Comment by luizaogs** [comment]
Though clicking on the archaeological identifier only takes me to the Op and not to the specific record (locus_deposit vs. locus_photo, for instance). Is that fine?

(I guess this pertains to #2304 )

**Comment by urapadmin** [comment]
> Though clicking on the archaeological identifier only takes me to the Op and not to the specific record (locus_deposit vs. locus_photo, for instance). Is that fine?
> 
> (I guess this pertains to [#2304](https://github.com/arch-kiosk/arch-kiosk-office/issues/2304) )

yes, that is at least intended.

**Comment by urapadmin** [comment]
> But then maybe don't call it "list archaeological entities DUG by a certain team member"? That's what tripped me up.

fair point.

**Comment by urapadmin** [comment]
> > But then maybe don't call it "list archaeological entities DUG by a certain team member"? That's what tripped me up.
> 
> fair point.

did that

**Comment by urapadmin** [comment]
@lbestock should also test if this is what she wanted

**Comment by luizaogs** [comment]
The title/description are a lot clearer now.

**Comment by lbestock** [comment]
Yes. Wow! It is what I wanted and it is immediately great and put to use.
